pkg/drivers/powerflex.go

```diff
@@ -440,3 +445,25 @@ func ValidateZonesInSecret(ctx context.Context, kube client.Client, namespace st
 
 	return nil
 }
+
+func RemoveVolume(configuration *v1.DaemonSetApplyConfiguration, volumeName string) error {
+	if configuration == nil {
+		return fmt.Errorf("RemoveVolume called with a nil daemonset")
+	}
+	podTemplate := configuration.Spec.Template
+	for i, vol := range podTemplate.Spec.Volumes {
+		if vol.Name != nil && *vol.Name == volumeName {
+			podTemplate.Spec.Volumes = append(podTemplate.Spec.Volumes[0:i], podTemplate.Spec.Volumes[i+1:]...)
+			break
+		}
+	}
+	for c := range podTemplate.Spec.Containers {
+		for i, volMount := range podTemplate.Spec.Containers[c].VolumeMounts {
+			if volMount.Name != nil && *volMount.Name == volumeName {
+				podTemplate.Spec.Containers[c].VolumeMounts = append(podTemplate.Spec.Containers[c].VolumeMounts[0:i], podTemplate.Spec.Containers[c].VolumeMounts[i+1:]...)
+				break
```

Comment:
This is not a big deal, but this `break`  will only interrupt the nested loop, you'll still remain in the containers loop. `return nil` or using break with a label would be an alternative.

Reply:
good catch, fixed